src/Entities/Entity.cpp

```diff
@@ -1032,40 +1032,18 @@ void cEntity::HandlePhysics(std::chrono::milliseconds a_Dt, cChunk & a_Chunk)
 	}
 
 	// Get water direction
-	Direction WaterDir = m_World->GetWaterSimulator()->GetFlowingDirection(BlockX, BlockY, BlockZ);
+	Vector3f WaterDir = m_World->GetWaterSimulator()->GetFlowingDirection(BlockX, BlockY, BlockZ, false);
```

Comment:
I would rule out this part of the code altogether, since my numerous experiments on the original server showed that the direction of the flow of the upper block does not affect the bottom one in any way.

Reply:
I'm guessing this is intended to handle two block tall mobs.  Have you tested entities in head height water flows?

If that is what it's for then the logic would make more sense directly in `cEntity::HandlePhysics` though.

Reply:
Yes, I agree, so I need to leave it, but in the code I did not encounter an appeal to this function, although it may be in the future.

Reply:
```cpp
Vector3f WaterDir;
for (int i = 0; i <= CeilC(GetHeight()); ++i)
{
```

Reply:
Yes, and this should be done in the entity class.